mllib/src/main/scala/org/apache/spark/ml/classification/LogisticRegression.scala

```diff
@@ -803,13 +829,17 @@ private class LogisticAggregator(
     this
   }
 
-  def count: Long = totalCnt
-
-  def loss: Double = lossSum / totalCnt
+  def loss: Double = {
+    require(weightSum >= 1.0, s"The effective number of samples should be " +
```

Comment:
We shouldn't define variance that way. `weight` is not necessarily representing how many instances. For example, the following definition would be compatible with the case when `weight === 1.0`:

~~~
Var[X] = E[X^2] - E[X]^2
  ~ (\sum_i w_i x_i^2 / sum_i w_i) - (\sum_i w_i x_i / sum_i w_i)^2
~~~

Though we still need to figure out how to make it compatible with unbiased version and implement it in a numerical stable way.